src/Entities/SplashPotionEntity.cpp

```diff
@@ -57,26 +57,21 @@ void cSplashPotionEntity::OnHitEntity(cEntity & a_EntityHit, Vector3d a_HitPos)
 
 void cSplashPotionEntity::Splash(Vector3d a_HitPos)
 {
-	m_World->ForEachEntity([=](cEntity & a_Entity)
+	double XZCalculation = 8.25/2;
+	double YCalculation = 4.25/2;
+	cBoundingBox SplashDistanceBox = cBoundingBox(a_HitPos.x - XZCalculation, a_HitPos.x + XZCalculation, a_HitPos.y - YCalculation, a_HitPos.y + YCalculation, a_HitPos.z - XZCalculation, a_HitPos.z + XZCalculation);
+
+	m_World->ForEachEntityInBox(SplashDistanceBox, [=](cEntity & a_Entity)
 		{
 			if (!a_Entity.IsPawn())
 			{
 				// Not an entity that can take effects
 				return false;
 			}
 
-			Vector3d SplashDistanceCube = (a_Entity.GetPosition() - a_HitPos);
-			double SplashDistance = SplashDistanceCube.Length();
-
-			// Wiki says 8.25 x 8.25 x 4.25 Cuboid.
-			if ((std::max(SplashDistanceCube.x, SplashDistanceCube.z)) > 8.25 || (SplashDistanceCube.y > 4.25))
-			{
-				// Too far away
-				return false;
-			}
-
 			// y = -0.25x + 1, where x is the distance from the player. Approximation for potion splash.
 			// TODO: better equation
+			double SplashDistance = (a_Entity.GetPosition() - a_HitPos).Length();
 			double Reduction = -0.25 * SplashDistance + 1.0;
 			Reduction = std::max(Reduction, 0.0);
```

Comment:
Hmm, this isn't your code, but calling AddEntityEffect with a zero reduction means the existing effect is removed and the new effect has 0 duration. This seems wrong although I can't be totally sure of the vanilla behaviour - could you confirm?

Reply:
Sure let me see if I can confirm. I assume that's why the original author put that todo in there.

Reply:
`For instant effects (i.e. Healing or Harming), the potency of the effect reduces linearly from 100% on a direct hit to 0% at 4 blocks' distance. For other effects, the potency is unchanged, but the duration decreases linearly on the same scale (rounded to the nearest 1⁄20 second), with no effect being applied if the duration would be 1 second or less. ` It does seem like a zero reduction is correct. Seems like the effect is not applied if it's 1 second or less.

Reply:
The issue is not that the effect won't be applied, but that hitting a player with a zero-effect splash potion will remove the existing effect - I think this may need a different function to CheckGreaterAndAddEntityEffect(), unless _that_ behaviour is vanilla as well.

Reply:
Ahh I see what you are saying. And I see that it is _not_ the way it is in vanilla. So I think that I need to implement this as suggested. Should it be fixed as part of this, or logged as a separate issue?

Reply:
Ok sorry for the spam. But when testing this behavior out, I don't seem to be getting the effect removed. So although it does look like via the code that is what should happen, but i'm not seeing it. Let me keep testing and playing around

Reply:
I think the right path here is to remove the TODO in this PR, as you've verified the calculation is correct, then open a new issue to fix the other problem, which may be more pervasive (we should check that other places where entity effects are added do not inadvertently remove existing effects).